modules/m_sasl.cpp

```diff
@@ -320,23 +320,43 @@ class ModuleSASL : public Module
 			IRCD->SendSASLMechanisms(mechs);
 	}
 
- public:
-	ModuleSASL(const Anope::string &modname, const Anope::string &creator) : Module(modname, creator, VENDOR),
-		sasl(this), plain(this), external(NULL)
+	template<typename Mech>
+	void ToggleMech(Mech *mech, bool enable)
 	{
-		try
+		if (enable)
+		{
+			try
+			{
+				mech = new Mech(this);
```

Comment:
Are you sure about this? I don't think it would change the pointer in the module unless `Mech` is a reference?